source/app.d

```diff
@@ -210,6 +288,93 @@ void updateGithubComment(string action, IssueRef[] refs, Issue[] descs, string c
     }
 }
 
+
+//==============================================================================
+// Github Auto-merge
+//==============================================================================
+
+alias LabelsAndCommits = Tuple!(Json[], "labels", Json[], "commits");
+alias LabelInfo = Tuple!(bool, "hasAutoMerge", bool, "hasAutoMergeSquash");
+
+auto analyseLabels(Json[] labels)
+{
+    auto labelNames = labels.map!`a["name"].get!string`.array;
+    bool hasAutoMerge = labelNames.any!`a == "auto-merge"`;
+    bool hasAutoMergeSquash = labelNames.any!`a == "auto-merge-squash"`;
+    return LabelInfo(hasAutoMerge, hasAutoMergeSquash);
+}
+
+auto handleGithubLabel(string repoSlug, uint pullRequestNumber, string commitsURL, bool isOpen)
+{
+    auto url = "%s/repos/%s/issues/%d/labels".format(githubAPIURL, repoSlug, pullRequestNumber);
+    auto res = ghGetRequest(url);
+
+    auto labels = res.readJson[];
+    auto labelInfo = analyseLabels(labels);
+    Json[] commits = null;
+
+    if (labelInfo.hasAutoMerge || labelInfo.hasAutoMergeSquash)
+    {
+        commits = getCommits(commitsURL);
+
+        if (!isOpen)
+        {
+            logWarn("Can't auto-merge PR %s/%d - it is already closed", repoSlug, pullRequestNumber);
+            goto end;
+        }
+
+        if (commits.length == 0)
+        {
+            logWarn("Can't auto-merge PR %s/%d has no commits attached", repoSlug, pullRequestNumber);
+            goto end;
+        }
+
+        string lastCommitSha = commits[$ - 1]["sha"].get!string;
+        string[string] reqInput = ["sha": lastCommitSha];
+
+        // lazy implementation -> try merge
+        if (labelInfo.hasAutoMerge)
+            reqInput["merge_method"] = "merge";
+
+        if (labelInfo.hasAutoMergeSquash)
+            reqInput["merge_method"] = "squash";
+
+        auto prUrl = "%s/repos/%s/pulls/%d/merge".format(githubAPIURL, repoSlug, pullRequestNumber);
+        ghSendRequest((scope req){
+            req.method = HTTPMethod.PUT;
+            // custom media type is required during preview period:
+            // https://developer.github.com/changes/2016-09-26-pull-request-merge-api-update/
+            req.headers["Accept"] = "application/vnd.github.polaris-preview+json";
+            req.writeJsonBody(reqInput);
+        }, prUrl);
```

Comment:
We try the merge here .. it will probably fail & in the error case we just print a log message